src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsServerRouter/DiagnosticsServerRouterFactory.cs

```diff
@@ -282,6 +286,149 @@ public void CreatedNewServer(EndPoint localEP)
         }
     }
 
+    /// <summary>
+    /// This class represent a WebSocket server endpoint used when building up router instances.
+    /// </summary>
+    internal class WebSocketServerRouterFactory : IIpcServerTransportCallbackInternal
```

Comment:
The factories implementing `IIpcServerTransportCallbackInternal` all have a lot of copy/pasted code, too

Reply:
What is the difference between this copy and TcpServerRouterFactory? Looks quite similar, wouldn't alternative be to subclass TcpServerRouterFactory and make small adjustments to the TcpServerRouterFactory making that possible?

Reply:
yea I'll think about sharing more with the TCP implementation.  The main difference is how the endpoints are described, I think (uri vs IPEndPoint)

Reply:
This isn't copy/pasted anymore